src/lib_modules.ml

```diff
@@ -63,14 +72,14 @@ let make_wrapped ~(lib : Dune_file.Library.t) ~dir ~transition ~modules
 
          https://github.com/ocaml/dune/issues/567 *)
       Some
-        (Module.make (Module.Name.add_suffix main_module_name "__")
+        (Module.make (Module.Name.add_suffix alias_module_name "__")
```

Comment:
Don't we end up adding 4 trailing underscores here?

Reply:
Looking at the tests, it seems that we don't. However, reading the code, it really seems to me that we should end up with 4 underscores. @rgrinberg do you agree?

Reply:
It's because `Module.Name.Map.mem modules main_module_name` is never true i suppose. It could be true in the case where the lib interface module is virtual however. I will fix this.

Reply:
Now the implementation case is handled separately and the separator is always `__`

Reply:
Thanks, that looks clearer.